src/components/base/BaseInput.vue

```diff
@@ -0,0 +1,211 @@
+<template>
+  <div
+    class="base-input"
+    :class="{
+      'has-text': !!modelValue,
```

Comment:
it's not critical, but i use Boolean(value), because in this case you have double conversions